catroid/src/main/java/org/catrobat/catroid/content/bricks/LegoNxtPlayToneBrick.java

```diff
@@ -102,6 +109,26 @@ public View getView(Context context, int brickId, BaseAdapter baseAdapter) {
 				.setTextFieldId(R.id.nxt_tone_duration_edit_text);
 		getFormulaWithBrickField(BrickField.LEGO_NXT_DURATION_IN_SECONDS).refreshTextField(view);
 
+		TextView times = (TextView) view.findViewById(R.id.brick_nxt_play_tone_seconds);
+
+		if (getFormulaWithBrickField(BrickField.LEGO_NXT_DURATION_IN_SECONDS).isSingleNumberFormula()) {
```

Comment:
this chunk of code appears again and again, can you please put it in a function that takes a textField and a brickField as inputs, and check if you can maybe move it in a higher hirarchy class of Brick, (probably FormulaBrick)

Reply:
done